File: leecode/1700-20221019/code.py
from collections import Counter
List = list


class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        stu_conter = Counter(students)
        for i in sandwiches:
            if stu_conter[i]:
                stu_conter[i] -= 1
                continue
            break
        return stu_conter[0]+stu_conter[1]

    def __call__(self, *args, **kwds):
        myans = self.countStudents(*args)
        self.test(myans, **kwds)

    def test(self, myans, ans):
        if myans != ans:
            print(f"F: {myans} >>> {ans}")
        else:
            print(f"T: {ans}")


Solution()([1, 1, 0, 0], [0, 1, 0, 1], ans=0)
Solution()([1, 1, 0, 0, 1], [0, 1, 0, 1], ans=1)
Solution()([1, 1, 0, 0], [0, 1, 0, 1, 1], ans=0)
Solution()([1, 1, 1, 0, 0, 1], [1, 0, 0, 0, 1, 1], ans=3)

File: leecode/1700-20221019/code.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <numeric>
using namespace std;
class Solution
{
public:
  int countStudents(vector<int> &students, vector<int> &sandwiches)
  {
    int s1 = accumulate(students.begin(), students.end(), 0);
    int s0 = students.size() - s1;
    int sand_len = sandwiches.size();
    for (int i = 0; i < sand_len; i++)
    {
      if (sandwiches[i] == 1 && s1 > 0)
        s1--;
      else if (sandwiches[i] == 0 && s0 > 0)
        s0--;
      else
        break;
    }
    return s0 + s1;
  }
};
int main()
{
  Solution solution;
  vector<int> students = {1, 0, 1, 0};
  vector<int> sandwitches = {1, 1, 0, 1};
  cout << solution.countStudents(students, sandwitches) << endl;
  return 0;
}